packages/app/src/cli/services/app/config/push.ts

```diff
@@ -0,0 +1,45 @@
+import {PushConfig, PushConfigSchema} from '../../../api/graphql/push_config.js'
+import {AppInterface} from '../../../models/app/app.js'
+import {ensureAuthenticatedPartners} from '@shopify/cli-kit/node/session'
+import {partnersRequest} from '@shopify/cli-kit/node/api/partners'
+import {AbortError} from '@shopify/cli-kit/node/error'
+import {renderSuccess} from '@shopify/cli-kit/node/ui'
+import {OutputMessage} from '@shopify/cli-kit/node/output'
+import {basename} from '@shopify/cli-kit/node/path'
+
+export interface Options {
+  app: AppInterface
+}
+
+export async function pushConfig(options: Options) {
+  const token = await ensureAuthenticatedPartners()
+  const mutation = PushConfig
+
+  const {configuration} = options.app
+  const configFileName = basename(options.app.configurationPath)
+
+  if (!configuration.client_id) {
+    abort(`${configFileName} does not contain a client_id.`)
```

Comment:
I think it is a good idea to return a more instructive message, perhaps something like "The repository must be linked to an app to run this command. Refer to [docs]() to learn more." 

However, trying to run `link` for unlinked repositories will break `dev` and `generate` commands support for legacy TOML files / force users to opt-in - although it would not be too strange to only implement this behaviour for `push` since `push` does not make sense out of a linked context.

Reply:
For sure, we're planning to get placeholder copy in and then jam with Jen on the final copy in a broader take.